packages/lit-html/src/directives/choose.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * Chooses and evaluates a template function from a list based on matching
+ * the given `value` to a case.
+ *
+ * Cases are structured as `[caseValue, func]`. `value` is matched to
+ * `caseValue` by strict equality. The first match is selected.
+ *
+ * This is similar to a switch statement, but as an expression and without
+ * fallthrough.
+ */
+export const choose = <T, V>(
+  value: T,
+  cases: Array<[T, () => V]>,
+  defaultCase?: () => V
+) => {
+  for (const c of cases) {
+    const caseValue = c[0];
+    const fn = c[1];
+    if (caseValue === value) {
```

Comment:
On phone right now but maybe use the compare function that's also used in lit-labs/task which can also compare NaN?

Reply:
We don't use `Object.is()` in core yet because it requires another polyfill in IE - which we still support. We could use it here without it technically being a breaking change, but it would be harder to describe which polyfills we require.

So I think that we probably don't want to do that and if you need to equality compare against numbers that might be `NaN` that you either convert the number to a sentinel first, or use some other control-flow construct. Maybe ternary before the `choose()`. It also seems to me like if you're dealign with `NaN` you might want to be using inequalities which would rule out `choose()` anyway.

These new helpers are simple and meant to cover many but not all use cases, and we hope people can write their own if they need to customize their behavior.

In the future I could see a `match()` directive that would allow for any comparison:

```ts
match(x, [
  [(x) => x < 0, html`negative`],
  [(x) => x > 0, html`positive`],
  [(x) => x === 0, html`zero`],
  [(x) => Object.is(x, Nan), html`oops`],
])
```

Reply:
I thought about https://github.com/lit/lit/blob/fcc2b3d0054e69e6f76588ea9f440117b6d0deed/packages/reactive-element/src/reactive-element.ts#L280 but I agree that this is probably overkill as this directive is mostly a convenience for the common use cases.